WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooShippingAddPackageView.swift

```diff
@@ -0,0 +1,254 @@
+import SwiftUI
+
+struct WooShippingAddPackageView: View {
+    enum PackageProviderType: CaseIterable {
+        case custom, carrier, saved
+        var name: String {
+            switch self {
+            case .custom:
+                return Localization.custom
+            case .carrier:
+                return Localization.carrier
+            case .saved:
+                return Localization.saved
+            }
+        }
+    }
+
+    @Environment(\.presentationMode) var presentationMode
+
+    @State var selectedPackageType = PackageProviderType.custom
+    var addPackageButtonDisabled: Bool {
+        for (_, value) in fieldValues {
+            if value.isEmpty {
+                return true
+            }
+        }
+        return fieldValues.count != WooShippingAddPackageDimensionView.DimensionType.allCases.count
+    }
+
+    var body: some View {
+        NavigationView {
+            VStack {
+                Picker("", selection: $selectedPackageType) {
+                    ForEach(PackageProviderType.allCases, id: \.self) {
+                        Text($0.name)
+                    }
+                }
+                .pickerStyle(.segmented)
+                .padding()
+                Spacer()
+                selectedPackageTypeView
+                Spacer()
+                Button(Localization.addPackage) {
+                    // add package
+                }
+                .buttonStyle(PrimaryButtonStyle())
+                .disabled(addPackageButtonDisabled)
+                .padding()
+            }
+            .toolbar {
+                ToolbarItem(placement: .topBarLeading) {
+                    Button(action: {
+                        presentationMode.wrappedValue.dismiss()
+                    }, label: {
+                        Text(Localization.cancel)
+                    })
+                }
+            }
+            .navigationTitle(Localization.addPackage)
+            .navigationBarTitleDisplayMode(.inline)
+        }
+        .navigationViewStyle(.stack)
+    }
+
+    @ViewBuilder
+    private var selectedPackageTypeView: some View {
+        ScrollView {
+            switch selectedPackageType {
+            case .custom:
+                customPackageView
+            case .carrier:
+                carrierPackageView
+            case .saved:
+                savedPackageView
+            }
+        }
+    }
+
+    enum PackageType: CaseIterable {
+        case box, envelope
+        var name: String {
+            switch self {
+            case .box:
+                return Localization.box
+            case .envelope:
+                return Localization.envelope
+            }
+        }
+    }
+
+    @State var packageType: PackageType = .box
+    @State var showSaveTemplate: Bool = false
+    @State var length: String = ""
+    @State var fieldValues: [WooShippingAddPackageDimensionView.DimensionType: String] = [:]
```

Comment:
nit: Could we have some documentation about what these properties correspond to? Some of them feel fairly clear but it took me a little more work to understand what fields `fieldValues` refers to.